controller/kubernetes_node_controller.go

```diff
@@ -322,11 +322,11 @@ func (knc *KubernetesNodeController) syncDefaultDisks(node *longhorn.Node) (err
 
 	node.Spec.Disks = disks
 
-	updatedNode, err := knc.ds.UpdateNode(node)
+	_, err = knc.ds.UpdateNode(node)
 	if err != nil {
 		return err
 	}
-	node = updatedNode
+
```

Comment:
Besides, I realized that Longhorn updates node CRs blindly and immediately in this controller. We will open a separate PR for this issue.

Reply:
The fix for this part is in PR https://github.com/longhorn/longhorn-manager/pull/1148